eZ/Publish/Core/MVC/Symfony/View/Builder/ContentViewBuilder.php

```diff
@@ -100,6 +100,12 @@ public function buildView(array $parameters)
         } elseif ($location instanceof Location) {
             // if we already have location load content true it so we avoid dual loading in case user does that in view
             $content = $location->getContent();
+            if (!$this->canRead($content, $location)) {
```

Comment:
We need to take embed / non embed into account, like done in loading logic implicitly.
Lets add a third argument to canRead(). `bool $isEmbed = true` and if false don't check  for `view_embed `

Reply:
Done in https://github.com/ezsystems/ezpublish-kernel/pull/2419/commits/7e2cae540ca0a25cb46512936f29065840f7f97c.